src/test/java/com/google/api/codegen/testdata/discogapic/java/java_simplecompute.v1.json.baseline

```diff
@@ -6480,3 +6485,333 @@ public class HttpJsonAddressStub extends AddressStub {
+               AddressSettings.defaultHttpJsonTransportProviderBuilder()
+                   .setHttpTransport(mockService).build()).build();
+    client =
+       AddressClient.create(clientSettings);
+  }
+
+  @After
+  public void cleanUp() {
+    mockService.reset();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    client.close();
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void deleteAddressTest() {
+    String httpErrorMessage = "httpErrorMessage1276263769";
+    String targetId = "targetId-815576439";
+    String kind = "kind3292052";
+    String description = "description-1724546052";
+    String statusMessage = "statusMessage-239442758";
+    String selfLink = "selfLink-1691268851";
+    String insertTime = "insertTime-103148397";
+    Integer httpErrorStatusCode = 1386087020;
+    String zone = "zone3744684";
+    String targetLink = "targetLink-2084812312";
+    String creationTimestamp = "creationTimestamp567396278";
+    String name = "name3373707";
+    Integer progress = 1001078227;
+    String operationType = "operationType-1432962286";
+    String startTime = "startTime-1573145462";
+    String endTime = "endTime1725551537";
+    String id = "id3355";
+    RegionName region = RegionName.of("[PROJECT]", "[REGION]");
+    String clientOperationId = "clientOperationId-239630617";
+    String user = "user3599307";
+    String status = "status-892481550";
+    Operation expectedResponse = Operation.newBuilder()
+      .setHttpErrorMessage(httpErrorMessage)
+      .setTargetId(targetId)
+      .setKind(kind)
+      .setDescription(description)
+      .setStatusMessage(statusMessage)
+      .setSelfLink(selfLink)
+      .setInsertTime(insertTime)
+      .setHttpErrorStatusCode(httpErrorStatusCode)
+      .setZone(zone)
+      .setTargetLink(targetLink)
+      .setCreationTimestamp(creationTimestamp)
+      .setName(name)
+      .setProgress(progress)
+      .setOperationType(operationType)
+      .setStartTime(startTime)
+      .setEndTime(endTime)
+      .setId(id)
+      .setRegion(region.toString())
+      .setClientOperationId(clientOperationId)
+      .setUser(user)
+      .setStatus(status)
+      .build();
+    mockService.addResponse(expectedResponse);
+    mockService.setSerializer(deleteAddressMethodDescriptor);
+
+    AddressName address = AddressName.of("[PROJECT]", "[REGION]", "[ADDRESS]");
+
+    Operation actualResponse =
+        client.deleteAddress(address);
+    Assert.assertEquals(expectedResponse, actualResponse);
+
+    List<String> actualRequests = mockService.getRequestPaths();
+    Assert.assertEquals(1, actualRequests.size());
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void deleteAddressExceptionTest() throws Exception {
+    ApiException exception = ApiExceptionFactory.createException(new Exception(), FakeStatusCode.of(Code.INVALID_ARGUMENT), false);
+    mockService.addException(exception);
+    mockService.setSerializer(deleteAddressMethodDescriptor);
```

Comment:
Instead of having a mock service and then setting serializers in it within tests, I would much prefer to generate API-specific mock services (like we do for grpc) which know what serializers to use.

Reply:
Ok removing the http_unit_test.snip and just using the existing test.snip for now.